### ISSUE TITLE
FileVersion in example appears to do nothing

### ISSUE THREAD

**Issue by starsoccer** [jet2jet/resedit-js#23]
The example code has a FileVersion specified, but as far as I can tell this does nothing. The true file version seems to come from the fileVersionMS and fileVersionLS. I think it would be useful to simply remove the FileVersion line to avoid confusion if it does indeed do nothing.

It would also be helpful to include a utility for converting a regular number the fileVersionMS & fileVersionLS formats as this does not seem to be straightforward

**Comment by jet2jet** [comment]
According to Windows version resource specification, there are two 'FileVersion' values (as well as 'ProductVersion' values). The one is in VS_FIXEDFILEINFO (represented as numeric value) and the other is contained as variable string value. In general, these should be semantically same values, but can be different. (In modern Windows, it seems that 'FileVersion' string value (and 'ProductVersion' numeric value) is not displayed.)

In resedit, since it treats raw resource data, there are the fields: `fixedInfo.fileVersionMS` / `fixedInfo.fileVersionLS` and `FileVersion` (set with `setStringValues`), as well as `fixedInfo.productVersionMS` / `fixedInfo.productVersionLS` / and `ProductVersion`. Currently you should set these fields correctly, whether or not they are actually displayed.

**Comment by jet2jet** [comment]
To reduce this confusings, I'm thinking to add utility methods to specify versions like:

```ts
    public setFileVersion(major: number, minor: number, micro?: number, revision?: number): void;
    public setProductVersion(major: number, minor: number, micro?: number, revision?: number): void;
```

**Comment by starsoccer** [comment]
Regarding the 2 FileVersion and ProductVersion fields, I was simply referring to the versions I can see in windows when right clicking and selecting details. I can see one of each field but as far as I can tell specifying the value in the example only does not affect whatever version is shown here.

Regarding utility functions I would personally make the lang field optional. I am not super familiar with what lang controls but I imagine in most cases users will just use whatever is the default or for regular(English numbering). For my use case in particular I am simply planning to basically just pull the package.json version number and set that as the FileVersion. This will come as a string since I believe that is how its stored in the package.json. Splitting it by periods shouldnt be a big deal.

**Comment by jet2jet** [comment]
I added `setFileVersion` and `setProductVersion` methods (in e40b2d3). In addition to the previous comment, both methods accepts string value and `lang` value can be omitted.

<details>
<summary>The declarations:</summary>

```ts
    /**
     * Sets 'FileVersion' property with specified values.
     * This methods writes `fixedInfo.fileVersionMS` and `fixedInfo.fileVersionLS` fields,
     * and writes `FileVersion` string with the value `<major>.<minor>.<micro>.<revision>`.
     * @param major The major version (clamped between 0 and 65535)
     * @param minor The minor version (clamped between 0 and 65535)
     * @param micro The micro version (clamped between 0 and 65535; default is 0)
     * @param revision The revision value (clamped between 0 and 65535; default is 0)
     * @param lang The language (default: this.lang -> picked from existings -> 1033)
     */
    setFileVersion(major: number, minor: number, micro?: number, revision?: number, lang?: number): void;
    /**
     * Sets 'FileVersion' property with specified values.
     * This methods writes `fixedInfo.fileVersionMS` and `fixedInfo.fileVersionLS` fields,
     * and writes `FileVersion` string with the value `<major>.<minor>.<micro>.<revision>`.
     * @param version The version string value (should be `x.x.x.x` format; each integer clamped between 0 and 65535)
     * @param lang The language (default: this.lang -> picked from existings -> 1033)
     */
    setFileVersion(version: string, lang?: number): void;

    /**
     * Sets 'ProductVersion' property with specified values.
     * This methods writes `fixedInfo.productVersionMS` and `fixedInfo.productVersionLS` fields,
     * and writes `ProductVersion` string with the value `<major>.<minor>.<micro>.<revision>`.
     * @param major The major version (clamped between 0 and 65535)
     * @param minor The minor version (clamped between 0 and 65535)
     * @param micro The micro version (clamped between 0 and 65535; default is 0)
     * @param revision The revision value (clamped between 0 and 65535; default is 0)
     * @param lang The language (default: this.lang -> picked from existings -> 1033)
     */
    setProductVersion(major: number, minor: number, micro?: number, revision?: number, lang?: number): void;
    /**
     * Sets 'ProductVersion' property with specified values.
     * This methods writes `fixedInfo.productVersionMS` and `fixedInfo.productVersionLS` fields,
     * and writes `ProductVersion` string with the value `<major>.<minor>.<micro>.<revision>`.
     * @param version The version string value (should be `x.x.x.x` format; each integer clamped between 0 and 65535)
     * @param lang The language (default: this.lang -> picked from existings -> 1033)
     */
    setProductVersion(version: string, lang?: number): void;
```
</details>

**Comment by starsoccer** [comment]
Awesome thanks, once its released can you just post here and Ill give it a try and then close this assuming everything works

**Comment by jet2jet** [comment]
I published v1.2.0. Feel free to report if something is wrong.

**Comment by starsoccer** [comment]
Awesome, confirmed its working will close this
